### PR TITLE
EHF Invoice and Allowances/Charges on Invoice/InvoiceLine/Price

### LINKED ISSUE #257: invoice rabatt on lines and invoice rabatt total
This has been integrated in FB

The following point is locations where rabatt should be implemented 
- [x] Creating outgoing invoices
- [x] Sending invoices to FB
- [x] Importing outgoing invoices from FB
- [x] Importing Incoming invoices from FB

### LINKED ISSUE #291: Testing EHF updating
- [x] Align to the right 'Beløp' in the field with tax%(Last before buttons)
- [x] Add new line/allowance/charge shouldn't reload page on a newly created invoice(for outgoing invoices)
- [x] MVA-amount breaks(two lines) when amount is greater than 999.99 NOK, should not break until 9 999 999.99 NOK
- [x] Line control when adding allowance/charge on invoice
- [x] When adding allowance/charge on invoice line can you select between 'linje' and 'pris', but the page is changing/jumping when you switch between those two. Should not jump like that.
- [x] Log(created, locked and sent by) sometimes breaks, should not break.
- [x] If you add allowance/charge line on invoice, and not add any values -> Missing AccountPlanID to insert_voucher_line() (add warning and do not bookkeep that allowance/charge) **EDIT:** Per Arnt's instructions, do not allow saving the invoice if required info is empty.
- [x] Add new line with allowance/charge should save invoice, then add new line(for incoming invoices)
- [x] Invoice reconciliations are not imported from fb to lodo(reasons are imported, but reconciliations are not) **EDIT:** fixed on FB
- [x] Oauth authorization does not look good (translations seem to be missing)  **EDIT:** fixed on FB
  skjermbilde 2016-10-30 kl 16 10 12

### LINKED ISSUE #293: Testing EHF updating 2
- [x] When adding allowance/charge on line level does the dropdown in front cover part of input field
- [x] Incoming invoices imported do all have 0000-00-00 as due date (**EDIT**: FB issue with payment means node)
- [x] Sending invoice with 15% VAT -> Error: Vat category with code: and rate: 15.00, does not exists

### PR DESCRIPTION
**Pull request by mladjo2505**
Closes #257
Closes #291 
Closes #293 

- [x] Add CRUD for Invoice level allowances/charges
- [x] Add allowances/aharges for InvoiceLine/Price
- [x] Build an EHF compatible invoice when sending from LODO
- [x] Receive an EHF invoice
- [x] Add allowances/charges to edit view for invoices
- [x] Bookkeep invoice level allowances/charges in journal as separate lines
- [x] Fix line control check for bookkept outgoing invoices
- [x] Add the ability to show/hide all allowance/charge info on invoice edit page
- [x] Translate to Norwegian

### CODE REVIEW

File: modules/allowancecharge/view/edit.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    includeinc('left');
+?>
+
+<form name="allowancecharge" action="<? print $MY_SELF ?>" method="post">
+<input type="hidden" name="AllowanceChargeID" value="<? print $row->AllowanceChargeID ?>">
+<? print $message ?>
+<table cellspacing="0">
+<thead>
+    <tr>
+        <th>Allowance/Charge register
```
Comment (mladjo2505) [inline modules/allowancecharge/view/edit.php]: Close th and translate to: "Rabatt/Kostnad register"

File: modules/allowancecharge/view/edit.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        <th colspan="2">
+</thead>
+<tbody>
+    <tr>
+        <td>AllowanceChargeID</td>
```
Comment (mladjo2505) [inline modules/allowancecharge/view/edit.php]: Just ID instead of AllowanceChargeID

File: modules/allowancecharge/view/edit.php [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        <td colspan="2"><? print $row->AllowanceChargeID ?></td>
+    </tr>
+    <tr>
+        <td>Aktiv</td>
+        <td colspan="2"><? print $_lib['form3']->checkbox(array('table'=>$db_table, 'field'=>'Active', 'pk'=>$row->AllowanceChargeID, 'tabindex' => $tabindex++, 'value'=>$row->Active)) ?></td>
+    </tr>
+    <tr>
+        <td>Charge indicator</td>
```
Comment (mladjo2505) [inline modules/allowancecharge/view/edit.php]: Type

File: modules/allowancecharge/view/edit.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        <td colspan="2"><? print $_lib['form3']->Generic_menu3(array('data'=>array('1'=>'Charge', '0'=>'Allowance'), 'table'=>$db_table, 'field'=>'ChargeIndicator', 'pk'=>$row->AllowanceChargeID, 'tabindex' => $tabindex++, 'value'=>$row->ChargeIndicator)); ?></td>
+    </tr>
+    <tr>
+        <td>Reason</td>
```
Comment (mladjo2505) [inline modules/allowancecharge/view/edit.php]: Årsak

File: modules/allowancecharge/view/edit.php [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        <td colspan="2"><? print $_lib['form3']->input(array('type'=>'text', 'table'=>$db_table, 'field'=>'Reason', 'pk'=>$row->AllowanceChargeID, 'tabindex'=>$tabindex++, 'value'=>$row->Reason)) ?></td>
+    </tr>
+    <tr>
+        <td>Amount</td>
```
Comment (mladjo2505) [inline modules/allowancecharge/view/edit.php]: Beløp

File: modules/allowancecharge/view/edit.php [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        <td colspan="2"><? print $_lib['form3']->input(array('type'=>'text', 'table'=>$db_table, 'field'=>'Amount', 'pk'=>$row->AllowanceChargeID, 'tabindex'=>$tabindex++, 'value'=>$_lib['format']->Amount($row->Amount))) ?>
+    </tr>
+    <tr>
+        <td>Resultat konto in</td>
```
Comment (mladjo2505) [inline modules/allowancecharge/view/edit.php]: inn instead of in

File: modules/allowancecharge/view/edit.php [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+        <td colspan="2"><? print $_lib['form3']->accountplan_number_menu(array('table'=>$db_table, 'field'=>'InAccountPlanID', 'pk'=>$row->AllowanceChargeID, 'tabindex'=>$tabindex++, 'value'=>$row->InAccountPlanID, 'type' => array(0 => 'result', 1 => 'balance'), 'required' => 1)) ?></td>
+    </tr>
+    <tr>
+        <td>MVA in</td>
```
Comment (mladjo2505) [inline modules/allowancecharge/view/edit.php]: inn instead of in

File: modules/allowancecharge/view/edit.php [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+        <td colspan="2"><? if($vat_in->Percent) { print "$vat_in->Percent%"; } ?></td>
+    </tr>
+    <tr>
+        <td>Resultat konto out</td>
```
Comment (mladjo2505) [inline modules/allowancecharge/view/edit.php]: ut instead of out

File: modules/allowancecharge/view/edit.php [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+        <td colspan="2"><? print $_lib['form3']->accountplan_number_menu(array('table'=>$db_table, 'field'=>'OutAccountPlanID', 'pk'=>$row->AllowanceChargeID, 'tabindex'=>$tabindex++, 'value'=>$row->OutAccountPlanID, 'type' => array(0 => 'result', 1 => 'balance'), 'required' => 1)) ?></td>
+    </tr>
+    <tr>
+        <td>MVA out</td>
```
Comment (mladjo2505) [inline modules/allowancecharge/view/edit.php]: ut instead of out

File: modules/allowancecharge/view/edit.php [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+        <td>AllowanceChargeID</td>
+        <td colspan="2"><? print $row->AllowanceChargeID ?></td>
+    </tr>
+    <tr>
+        <td>Aktiv</td>
+        <td colspan="2"><? print $_lib['form3']->checkbox(array('table'=>$db_table, 'field'=>'Active', 'pk'=>$row->AllowanceChargeID, 'tabindex' => $tabindex++, 'value'=>$row->Active)) ?></td>
+    </tr>
+    <tr>
+        <td>Charge indicator</td>
+        <td colspan="2"><? print $_lib['form3']->Generic_menu3(array('data'=>array('1'=>'Charge', '0'=>'Allowance'), 'table'=>$db_table, 'field'=>'ChargeIndicator', 'pk'=>$row->AllowanceChargeID, 'tabindex' => $tabindex++, 'value'=>$row->ChargeIndicator)); ?></td>
```
Comment (mladjo2505) [inline modules/allowancecharge/view/edit.php]: Replace

``` php
'data'=>array('1'=>'Charge', '0'=>'Allowance')
```

with

``` php
'data'=>array('1'=>'Kostnad', '0'=>'Rabatt')
```

File: modules/allowancecharge/view/list.php [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+$query = "select ac.*, a1.AccountName as OutAccountName, a2.AccountName as InAccountName from $db_table as ac left join accountplan a1 on ac.OutAccountPlanID = a1.AccountPlanID left join accountplan a2 on ac.InAccountPlanID = a2.AccountPlanID";
+$result2 = $_lib['db']->db_query($query);
+$db_total = $_lib['db']->db_numrows($result2);
+
+?>
+<html xmlns="http://www.w3.org/1999/xhtml">
+<? print $_lib['sess']->doctype ?>
+<head>
+    <title>Empatix - allowance/charge list</title>
```
Comment (mladjo2505) [inline modules/allowancecharge/view/list.php]: rabatt/kostnad liste

File: modules/allowancecharge/view/list.php [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+<body>
+<?
+    includeinc('top');
+    includeinc('left');
+?>
+
+    <table class="lodo_data" width="700px">
+        <thead>
+            <tr>
+                <th>Allowance/Charge listen:
```
Comment (mladjo2505) [inline modules/allowancecharge/view/list.php]: Rabatt/Kostnad listen

File: modules/allowancecharge/view/list.php [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+            <tr>
+                <th style="text-align: right;">
+                  <? if($_lib['sess']->get_person('AccessLevel') >= 2) { ?>
+                    <form name="edit" action="<? print $_lib['sess']->dispatch ?>t=allowancecharge.edit" method="post">
+                        <? print $_lib['form3']->Input(array('type'=>'text', 'table'=>$db_table, 'field'=>'AllowanceChargeID')) ?>
+                        <? print $_lib['form3']->Input(array('type'=>'submit', 'name'=>'action_allowance_new', 'value'=>'Nytt allowance')) ?>
```
Comment (mladjo2505) [inline modules/allowancecharge/view/list.php]: Ny rabatt

File: modules/allowancecharge/view/list.php [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+                        <? print $_lib['form3']->Input(array('type'=>'submit', 'name'=>'action_charge_new', 'value'=>'Nytt charge')) ?>
```
Comment (mladjo2505) [inline modules/allowancecharge/view/list.php]: Ny kostand

File: modules/fakturabank/model/fakturabank.class.php [hunk 15]
```diff
@@ -654,6 +678,33 @@ private function validate_outgoing($invoicesO) {
...
+                    //   $ac->TaxCategory->Percent = 25.0;
+                    //   $ac->TaxCategory->ID = 'S';
+                    //   $InvoiceO->AllowanceCharge[] = $ac;
+                    // }
+                    foreach ($InvoiceO->AllowanceCharge as $allowance_charge) {
+                      $query = "select * from allowancecharge where ChargeIndicator = " . $allowance_charge->ChargeIndicator . " and lower(Reason) = lower('" . $allowance_charge->AllowanceChargeReason . "') and Active = 1";
+                      $allowance_charge_exists = $_lib['db']->get_row(array('query' => $query, 'debug' => false));
+                      if(!$allowance_charge_exists->AllowanceChargeID) {
+                        if ($allowance_charge->ChargeIndicator) {
+                          $InvoiceO->Status     .= "Charge: '" . $allowance_charge->AllowanceChargeReason . "' eksisterer ikke. ";
```
Comment (mladjo2505) [inline modules/fakturabank/model/fakturabank.class.php]: Kostnad:

File: modules/fakturabank/model/fakturabank.class.php [hunk 16]
```diff
@@ -654,6 +678,33 @@ private function validate_outgoing($invoicesO) {
...
+                        } else {
+                          $InvoiceO->Status     .= "Allowance: '" . $allowance_charge->AllowanceChargeReason . "' eksisterer ikke. ";
```
Comment (mladjo2505) [inline modules/fakturabank/model/fakturabank.class.php]: Rabatt:

File: modules/fakturabank/model/fakturabank.class.php [hunk 17]
```diff
@@ -892,6 +943,33 @@ private function validate_incoming($invoicesO) {
...
+                    //   $ac->TaxCategory->Percent = 25.0;
+                    //   $ac->TaxCategory->ID = 'S';
+                    //   $InvoiceO->AllowanceCharge[] = $ac;
+                    // }
+                    foreach ($InvoiceO->AllowanceCharge as $allowance_charge) {
+                      $query = "select * from allowancecharge where ChargeIndicator = " . $allowance_charge->ChargeIndicator . " and lower(Reason) = lower('" . $allowance_charge->AllowanceChargeReason . "') and Active = 1";
+                      $allowance_charge_exists = $_lib['db']->get_row(array('query' => $query, 'debug' => false));
+                      if(!$allowance_charge_exists->AllowanceChargeID) {
+                        if ($allowance_charge->ChargeIndicator) {
+                          $InvoiceO->Status     .= "Charge: '" . $allowance_charge->AllowanceChargeReason . "' eksisterer ikke. ";
```
Comment (mladjo2505) [inline modules/fakturabank/model/fakturabank.class.php]: Kostnad:

File: modules/fakturabank/model/fakturabank.class.php [hunk 18]
```diff
@@ -892,6 +943,33 @@ private function validate_incoming($invoicesO) {
...
+                        } else {
+                          $InvoiceO->Status     .= "Allowance: '" . $allowance_charge->AllowanceChargeReason . "' eksisterer ikke. ";
```
Comment (mladjo2505) [inline modules/fakturabank/model/fakturabank.class.php]: Rabatt:

File: modules/fakturabank/model/fakturabank.class.php [hunk 19]
```diff
@@ -1680,500 +1894,333 @@ public function hash_to_xml($InvoiceO) {
...
+                                $id_attributes = array('schemeID' => 'UNCL5305');
+                                self::createElementIfNotEmpty($doc, $category, 'cbc:ID', $Vat->TaxSubtotal->TaxCategory->ID, $id_attributes);
+                                self::createElementIfNotEmpty($doc, $category, 'cbc:Percent', $Vat->TaxSubtotal->TaxCategory->Percent);
+                                self::createElementIfNotEmpty($doc, $category, 'cbc:TaxExemptionReason', $Vat->TaxSubtotal->TaxCategory->TaxExemptionReason);
+                                $scheme = self::createElementIfNotEmpty($doc, $category, 'cac:TaxScheme');
+                                    self::createElementIfNotEmpty($doc, $scheme, 'cbc:ID', $Vat->TaxSubtotal->TaxCategory->TaxScheme->ID);
+                    }
                 }
+            } else {
+                print "TAX info mangler<br>\n";
```
Comment (mladjo2505) [inline modules/fakturabank/model/fakturabank.class.php]: Skatte

File: modules/invoicein/model/invoicein.class.php [hunk 20]
```diff
@@ -296,6 +376,12 @@ public function journal() {
                         $VoucherH['voucher_ForeignConvRate']    = (float)$InvoiceO->ForeignConvRate; //exchange::getConversionRate($InvoiceO->DocumentCurrencyCode);
                     }
 
+                    // Add Allowances/Charges on invoice level
```
Comment (mladjo2505) [inline modules/invoicein/model/invoicein.class.php]: Remove?

File: modules/invoicein/view/edit.php [hunk 21]
```diff
@@ -283,21 +532,60 @@
...
+    </tr>
+    <tr>
+        <td colspan="5" align="right">Rabatt sum</td>
+        <td colspan="2" id="invoiceout.T2" align="right"><? print $_lib['format']->Amount($sum_allowance) ?></td>
     </tr>
     <tr>
-        <td colspan="6" align="right">Sum MVA</td>
-        <td align="right"><nobr><? print $_lib['format']->Amount($vatlines) ?></nobr></td>
+        <td colspan="5" align="right">Kostnad sum</td>
+        <td colspan="2" id="invoiceout.T3" align="right"><? print $_lib['format']->Amount($sum_charge) ?></td>
```
Comment (mladjo2505) [inline modules/invoicein/view/edit.php]: Remove placeholder ids T1, T2 and T3

File: modules/invoicein/model/invoicein.class.php [hunk 22]
```diff
@@ -351,24 +469,29 @@ public function journal() {
...
                             // $TotalForeignPrice = round(($TotalForeignPrice * (($line->Vat/100) +1)), 2);
                             // We already have tax amount given to us by fakturabank.
                             // So why recalculate, and possibly make 0.01 error in recalculation?
                             // Just add them up and get the correct amount.
                             if ($VoucherH['voucher_ForeignCurrencyID'] != '') {
-                              $TotalForeignPrice = round(($TotalForeignPrice + $line->TaxAmount), 2);
-                              $TotalPrice = round(($TotalPrice + exchange::convertToLocal($VoucherH['voucher_ForeignCurrencyID'], $line->TaxAmount)), 2);
+                              $TotalForeignPrice = round(($TotalForeignPrice), 2);
+                              $TaxAmount = (!empty($line->TaxAmount)) ? $line->TaxAmount : $TotalForeignPrice * $line->Vat / 100.0;
+                              $TotalPrice = round(($TotalPrice + exchange::convertToLocal($VoucherH['voucher_ForeignCurrencyID'], $TaxAmount)), 2);
```
Comment (mladjo2505) [inline modules/invoicein/model/invoicein.class.php:500]: Remove line 488.
Test for foreign invoice incoming, if it is bookkept correctly, if not this is to blame.

Possibly replace this:

``` php
$TotalPrice = round(($TotalPrice + exchange::convertToLocal($VoucherH['voucher_ForeignCurrencyID'], $TaxAmount)), 2);
```

with:

``` php
$TotalPrice = round((exchange::convertToLocal($VoucherH['voucher_ForeignCurrencyID'], $TotalForeignPrice) + exchange::convertToLocal($VoucherH['voucher_ForeignCurrencyID'], $TaxAmount)), 2);
```

File: modules/allowancecharge/view/edit.php [hunk 23]
```diff
@@ -26,9 +28,9 @@
 ?>
 
 <?
-    if(!$vat_in) print "<div class='warning'>Feil inng&aring;ende konto valg</div>";
-    if(!$vat_out) print "<div class='warning'>Feil utg&aring;ende konto valg</div>";
-    print "<br>";
+    if ($message = $_lib['message']->get()) {
+      print "<div class='warning'>" . $_lib['message']->get() . "</div><br>";
```
Comment (mladjo2505) [inline modules/allowancecharge/view/edit.php]: `$message` instead of `$_lib['message']->get()`

File: modules/invoice/view/edit.php [hunk 24]
```diff
@@ -286,8 +283,14 @@ function updateInvoiceData() {
     }
   }
   var invoice_allowances_charges = document.getElementsByClassName('global_invoice_allowancecharge');
+
+  document.getElementById('invoice_errors').innerHTML = "";
+  var allowance_charge_errors = "";
+
   for(i = 0; i < invoice_allowances_charges.length; i++) {
     var allowance_charge_id = invoice_allowances_charges[i].id.split('_')[3];
+    var allowance_charge_id_id = document.getElementById('<? print $db_table4 ?>.AllowanceChargeID.'+allowance_charge_id).value;
```
Comment (mladjo2505) [inline modules/invoice/view/edit.php]: rename to clarify first is invoice related the other is from the register of allowances and charges.

File: modules/fakturabank/model/fakturabank.class.php [hunk 25]
```diff
@@ -178,6 +178,21 @@ private function find_project_by_id($ProjectID) {
...
+    private function find_car_by_code($CarCode) {
+        global $_lib;
+
+        $query = "SELECT `CarCode` FROM car WHERE `CarCode` = '$CarCode'";
+        $result = $_lib['storage']->db_query3(array('query' => $query));
+        if (!$result) {
+            return false;
+        }
+        if ($obj = $_lib['storage']->db_fetch_object($result)) {
+            return $CarCode;
```
Comment (mladjo2505) [inline modules/fakturabank/model/fakturabank.class.php]: should return CarID

File: modules/fakturabank/model/fakturabank.class.php [hunk 26]
```diff
@@ -741,6 +756,18 @@ private function extractIncomingAccountingCost(&$InvoiceO) {
                 }
             }
 
+            if ($acc_cost_params['carcode'] != '') {
```
Comment (mladjo2505) [inline modules/fakturabank/model/fakturabank.class.php]: Change accordingly.

File: modules/invoice/view/edit.php [hunk 27]
```diff
@@ -900,10 +900,20 @@ function markRed(elements) {
       <td class="menu"></td>
     </tr>
   <?
+      $allowances_charges = array();
```
Comment (mladjo2505) [inline modules/invoice/view/edit.php]: Use existing query from line 96

File: modules/invoicein/view/edit.php [hunk 28]
```diff
@@ -284,10 +284,20 @@ function updateAndPerformAction(link_button) {
       <td class="menu"></td>
     </tr>
   <?
+      $allowances_charges = array();
```
Comment (mladjo2505) [inline modules/invoicein/view/edit.php]: Use existing query on line 57

**Comment by mladjo2505** [comment]
New requests:

- [x] Allowance/charge register should have no incoming account information but should use the suppliers bookkeeping account for bookkeeping these allowances/charges
- [x] Allowance/charge register should have project and department
- [x] Allowances/charges should look like invoice lines(use the same headers) but should go after all invoice lines(similar to how the view is in fakturabank)
- [x] Charge should show with positive numbers and allowance should show with negative numbers

File: modules/fakturabank/model/fakturabank.class.php [hunk 29]
```diff
@@ -1003,20 +1003,30 @@ private function validate_incoming($invoicesO) {
...
                     }
-                    else {
-                      $InvoiceO->Status .= "Bil: " . $line->Item->AdditionalItemProperty->Value . " eksisterer ikke. ";
-                      $InvoiceO->Journal = false;
-                      $InvoiceO->Class   = 'red';
+                    foreach($line->Item->AdditionalItemProperty as $additional_item_property) {
+                      if ($additional_item_property->Name == 'Car') {
+                        includelogic("car/car");
+                        $query = "select * from car where CarCode='" . $additional_item_property->Value . "' and ". car::car_active_sql("car.CarID", $InvoiceO->IssueDate) ."=1";
+                        print $query;
```
Comment (Baxxx) [inline modules/fakturabank/model/fakturabank.class.php]: Remove this

**Comment by sls-lnluksa** [comment]
First round review: Nikola Lukovic (sls-lnluksa)
Final review with: @stoivo

**Comment by stoivo** [comment]
- [x] Check why we have linecontroll
- [x] Initial load of invoice does not print errors related to account plan not being selected correctly on invoice level allowances/charges

File: modules/invoicein/view/edit.php [hunk 30]
```diff
@@ -261,121 +263,34 @@ function updateAndPerformAction(link_button) {
...
         $vat_charge    += $vat_charge_tax_amount;
         $sum_allowance += $sum_allowance_taxable_amount;
         $sum_charge    += $sum_charge_taxable_amount;
-        $tax_categories[$acrow->VatPercent]->TaxableAmount += $sum_charge_taxable_amount - $sum_allowance_taxable_amount;
-        $tax_categories[$acrow->VatPercent]->TaxAmount     += $vat_charge_tax_amount - $vat_allowance_tax_amount;
-        if(is_null($allowances_charges[$acrow->AllowanceChargeID]["InVatID"])) {
+        $tax_categories[$acrow->VatPercent]->TaxableAmount += $sum_charge_taxable_amount + $sum_allowance_taxable_amount;
+        $tax_categories[$acrow->VatPercent]->TaxAmount     += $vat_charge_tax_amount + $vat_allowance_tax_amount;
+        if($acrow->AllowanceChargeID && is_null($allowances_charges[$acrow->AllowanceChargeID]["InVatID"])) {
           $ac_errors[] = "Feil inng&aring;ende konto valg f&oslash;r ". ($allowances_charges[$acrow->AllowanceChargeID]["charge_indicator"]?"kostnad":"rabatt") ." <a href='". $_lib['sess']->dispatch ."t=allowancecharge.edit&AllowanceChargeID=". $acrow->AllowanceChargeID ."'>". $acrow->AllowanceChargeID ."</a>";
```
Comment (stoivo) [inline modules/invoicein/view/edit.php]: Leverandørs resultat konto har ikke valgt Mva kode

File: modules/fakturabank/model/fakturabank.class.php [hunk 31]
```diff
@@ -1003,20 +1003,29 @@ private function validate_incoming($invoicesO) {
...
+                    if(!is_array($line->Item->AdditionalItemProperty)) {
+                      $line->Item->AdditionalItemProperty = array($line->Item->AdditionalItemProperty);
                     }
-                    else {
-                      $InvoiceO->Status .= "Bil: " . $line->Item->AdditionalItemProperty->Value . " eksisterer ikke. ";
-                      $InvoiceO->Journal = false;
-                      $InvoiceO->Class   = 'red';
+                    foreach($line->Item->AdditionalItemProperty as $additional_item_property) {
+                      if ($additional_item_property->Name == 'Car') {
+                        includelogic("car/car");
```
Comment (mladjo2505) [inline modules/fakturabank/model/fakturabank.class.php]: move outside of for loop

**Comment by mladjo2505** [comment]
@stoivo We should renumber the migration if it conflicts with an already existing one on master.

**Comment by stoivo** [comment]
Can you do that @mladjo2505

**Pull request merged**